bokeh/client/session.py

```diff
@@ -153,11 +153,16 @@ def __init__(self, session_id=DEFAULT_SESSION_ID, io_loop=None, url=DEFAULT_SERV
         self._connection = ClientConnection(session=self, io_loop=io_loop, url=url)
 
         self._current_patch = None
+        self._callbacks = []
 
     def _attach_document(self, document):
         self._document = document
         self._document.on_change(self._document_changed)
 
+    def add_periodic_callback(self, cb, callback_time):
```

Comment:
Ah, right. It's a document only thing and session don't care. tnx!

Regarding the implementation.. yes, I'm defining those on the document level, that is ioloop agnostic and just add/remove the `callback` it's list of callbacks and then triggers the related event. Session manages the events doing the right action on the ioloop.